### PR TITLE
Restart comments model setup

### PR DESCRIPTION
**Pull request by PaulHowley**

### CODE REVIEW

File: app/views/articles/_comment.html.erb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<div>
```
Comment (krisquigley) [inline app/views/articles/_comment.html.erb]: Is this div needed?

File: app/views/comments/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<h3>Post a Comment</h3>
+<%= form_for :comment do |f| %>
```
Comment (krisquigley) [inline app/views/comments/_form.html.erb]: This should be @comment, but you will probably need `[@article, @comment]` as comment is a relation of article

File: config/routes.rb [hunk 3]
```diff
@@ -16,12 +16,13 @@
   get 'projects', to: 'static_pages#projects'
 
   resources :articles, only: [:index, :show] do
-    # resources :comments
+    resources :comments, only: [:show, :create]
```
Comment (krisquigley) [inline config/routes.rb]: Do you need a show action?  I would say you need :new and :create

File: app/views/comments/_form.html.erb (hunk 2 above)
Comment (krisquigley) [reply app/views/comments/_form.html.erb]: You'll also need to instantiate @comment in your articles_controller

Comment (PaulHowley) [reply app/views/comments/_form.html.erb]: Hi Kris, I have @comment defined in a CommentsController so does it need to be in the articles_controller aswell?

File: config/routes.rb (hunk 3 above)
Comment (PaulHowley) [reply config/routes.rb]: Hi Kris, is :create & :new not the same thing?

File: app/views/comments/_form.html.erb (hunk 2 above)
Comment (krisquigley) [reply app/views/comments/_form.html.erb]: The comment controller will only be called when a comment fails validation.  I assume the comment form is inline on the article show view?

File: config/routes.rb (hunk 3 above)
Comment (krisquigley) [reply config/routes.rb]: No, new and create are 2 different actions

Comment (krisquigley) [reply config/routes.rb]: New instantiates the object, create saves it

File: app/views/comments/_form.html.erb (hunk 2 above)
Comment (PaulHowley) [reply app/views/comments/_form.html.erb]: I'm still struggling to get my head round this one.  Firstly I've inserted the form_for [@article, @comment], but it come up with the error -"First argument in form cannot contain nil or be empty".  Also I'm not sure how to instantiate @comment in the articles_controller.  I already have def create.. & def new.. in the comments controller so where do you intsantiate it? under  def show... ?

Comment (PaulHowley) [reply app/views/comments/_form.html.erb]: I think I've worked it out Kris. I have done @comment = Comment.new within the def show... method and the error has gone and articles & comments now both showing.  I'll start working on the tests and send a PR when ready. Cheers.

File: spec/features/comments_spec.rb [hunk 4]
```diff
@@ -1,19 +1,20 @@
 require 'rails_helper'
 
-Rspec.describe "comments", type: :feature do
+RSpec.describe "comments", type: :feature do
   describe "Adding a comment" do
```
Comment (krisquigley) [inline spec/features/comments_spec.rb]: You'll also need a spec for if the form fails validations, the user should get an error message if they don't enter commenter or body

File: app/views/articles/show.html.erb [hunk 5]
```diff
@@ -5,20 +5,18 @@
   <h2 class="c-article-container__title "><%= @article.title %></h2>
   <p class="c-article-container__text"><%= @article.text %></p>
 </div>
-<%= link_to 'Articles list', articles_path, class: "o-btn c-articles-btn" %>
-
-<p><strong>Commenter:</strong><%= comment.commenter %></p>
+<h3>Comments</h3>
+  
+  <% @article.comments.each do |comment| %>
```
Comment (krisquigley) [inline app/views/articles/show.html.erb]: indentation below for each line looks a bit off and space after <h3

File: app/views/comments/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<h3>Post a Comment</h3>
+
+<%= render partial: 'shared/errors', locals: {obj: @comment} %>
+<%= form_for [@article, @comment] do |f| %>
+  <p>
+    <%= f.label :commenter %><br/>
```
Comment (krisquigley) [inline app/views/comments/_form.html.erb]: do not use `<br/>` for layout, use css instead

File: config/routes.rb [hunk 7]
```diff
@@ -14,14 +14,16 @@
   get 'about', to: 'static_pages#about'
   get 'contact', to: 'static_pages#contact'
   get 'projects', to: 'static_pages#projects'
+  get 'articles/index'
```
Comment (krisquigley) [inline config/routes.rb]: this is already defined below

File: config/routes.rb [hunk 8]
```diff
@@ -14,14 +14,16 @@
...
 
   resources :articles, only: [:index, :show] do
-    # resources :comments
+    resources :comments
```
Comment (krisquigley) [inline config/routes.rb]: Restrict this to only the actions you need

**Pull request merged**